Extensions/DialogueTree/dialoguetools.js

```diff
@@ -0,0 +1,538 @@
+
+  if (this.dialogueText) {
+    this.clipTextEnd += 1;
+  }
+};
+
+// Condition to check if text scrolling has completed.
+// Use this action to prevent the user from skipping to next line before the current one has been printed fully
+gdjs.dialogueTree.cippedTextScrollingHasCompleted = function() {
+  if (this.dialogueData && this.dialogueText.length) {
+    return this.clipTextEnd >= this.dialogueText.length;
+  }
+  return false;
+};
+
+// Expression to get the current dialogue line with a scrolling effect (recommended)
+// Use this with the scrollCippedText action to achieve a classing scrolling text,as well as any <<wait>> effects to pause scrolling
+gdjs.dialogueTree.getClippedLineText = function() {
+  return this.dialogueText.length
+    ? this.dialogueText.substring(0, this.clipTextEnd)
+    : "";
+};
+
+// Expression to get the current complete dialogue line without using any scrolling effects.
+// Note that using this instead getClippedLineText will skip any <<wait>> commands entirely.
+gdjs.dialogueTree.getLineText = function() {
+  this.clipTextEnd = this.dialogueText.length;
+  return this.dialogueText.length ? this.dialogueText : "";
+};
+
+// Expression to get the number of command parameters in a command with parameters that has been caught by a commandIsCalled condition
+gdjs.dialogueTree.commandParametersCount = function() {
+  if (this.commandParameters && this.commandParameters.length > 1) {
+    return this.commandParameters.length - 1;
+  }
+  return 0;
+};
+
+/**
+ * Expression to get a command parameter in any command with parameters that has been caugth by a commandIsCalled condition
+ * @param {number} paramIndex The index of the parameter you want to get. If left empty, the expression will get the first parameter by default.
+ */
+gdjs.dialogueTree.getCommandParameter = function(paramIndex) {
+  if (
+    this.commandParameters &&
+    this.commandParameters.length >= paramIndex + 1
+  ) {
+    var returnedParam = this.commandParameters[paramIndex + 1];
+    return returnedParam ? returnedParam : "";
+  }
+  return "";
+};
+
+/**
+ * Condition to catch <<commands>> and <<commands with parameters>> from the current Dialogue Line.
+ * Use this to trigger events in the event sheet that relate to the story you are telling throught the dialogue.
+ * @param {string} command The command you want to check for being called. Write it without the <<>>
+ */
+gdjs.dialogueTree.commandIsCalled = function(command) {
+  var { commandCalls, clipTextEnd, dialogueText } = gdjs.dialogueTree;
+
+  if (this.pauseScrolling || !commandCalls) return false;
+  return this.commandCalls.some(function(call, index) {
+    if (clipTextEnd < call.time) return false;
+    if (call.cmd === "wait" && clipTextEnd !== dialogueText.length) {
+      gdjs.dialogueTree.pauseScrolling = true;
+      setTimeout(function() {
+        gdjs.dialogueTree.pauseScrolling = false;
+        commandCalls.splice(index, 1);
+      }, parseInt(call.params[1]));
```

Comment:
Always set `, 10` parameter when calling parseInt, otherwise you'll get funny results when entering a number starting with 0 (and people will do that :))